File: Euclid/Network/Streams/Util/DelimeterEntry.cs
namespace Euclid.Network.Streams.Util
{
    public class DelimeterEntry
    {
        #region Private variables

        private string m_Value;
        private string m_Delimiter;

        #endregion

        #region Fields

        public string Value
        {
            get { return m_Value; }
        }

        public string Delimiter
        {
            get { return m_Delimiter; }
        }

        #endregion

        #region Constructor

        public DelimeterEntry(object value, object delimeter)
        {
            m_Value = (value == null ? string.Empty : value).ToString();
            m_Delimiter = (delimeter == null ? string.Empty : delimeter).ToString();
        }

        #endregion
    }
}

File: euclid/src/main/java/euclid/network/streams/util/DelimeterEntry.java
package euclid.network.streams.util;

public final class DelimeterEntry {

    private final String value;
    private final String delimiter;

    public DelimeterEntry(Object value, Object delimeter) {
        this.value = (value == null ? "" : value.toString());
        this.delimiter = (delimeter == null ? "" : delimeter.toString());
    }

    public String getValue() {
        return value;
    }

    public String getDelimiter() {
        return delimiter;
    }
}
